### PR TITLE
Added coding standards, and merged in a couple of other PRs

### PR DESCRIPTION
**Pull request by woubuc**
I added a `.editorconfig` file and added some guidelines and coding standards to the readme. I've tried to keep it beginner-friendly, and not too restrictive, but of course this is a starting point and I welcome a discussion around which standards we should follow.

Even with a "freeform" project like this one, I think it wouldn't hurt to have some coding standards in place, because it will make it easier for others to pick up and join in at any point.

Some other things:
- Cleaned up the code
- Small css tweaks
- Merged in @srebalaji's changes (#20)
- Separated XMLHttpRequest code to be a bit more DRY
- Replaced the png image of the PR icon with an svg version
- Merged in @nrobinson2000's footer (#22)
- Made the links on the page itself open in a new window
- Merged in @grimreaper's license (#14)
- Merged in @grimreaper's CI tests (#15)
- Merged in @dakebl's changes (#24)

Check it out at https://woubuc.github.io/1pr/

### CODE REVIEW

**Comment by bc-AlyssNoland** [comment]
You should add a .gitignore for your editor file :)

**Comment by woubuc** [comment]
@bc-AlyssNoland which file do you mean? The `.editorconfig` file is supposed to be there.

**Comment by bc-AlyssNoland** [comment]
@woubuc gotcha. I wasn't sure since that looked like an editor flavor akin to a webstorm file

**Comment by nrobinson2000** [comment]
@woubuc I really like what you are doing with the navigation bar in this PR.

**Comment by nrobinson2000** [comment]
I think you should also merge the Unit testing @grimreaper is working on. https://github.com/thepracticaldev/1pr/pull/15

**Comment by dakebl** [comment]
Need to style the footer section to compensate content clipping @nrobinson2000 @woubuc  
image

Might be worth putting the rest of the content in a div wrapper and padding out the bottom.

File: index.html [hunk 1]
```diff
@@ -47,7 +72,27 @@
 
 	</div>
```
Comment (nrobinson2000) [inline index.html]: I think you should add a `<br>` or two here to ensure that the footer doesn't overlap the contributor images on mobile.

Comment (nrobinson2000) [inline index.html]: Wow. I posted this at the exact same time as @dakebl

Comment (dakebl) [inline index.html]: I would suggest styling the ``contributors-container`` with the appropriate margin/padding, rather than using ``<br />`` tags.

Comment (nrobinson2000) [inline index.html]: Well yes, `<br>` would be a temporary fix and padding is much better.

**Comment by woubuc** [comment]
Merged in the license and fixed the footer issue

**Comment by nrobinson2000** [comment]
How about unit testing?

File: index.html (hunk 1 above)
Comment (woubuc) [inline index.html]: @nrobinson2000 Using `<br>` for anything but blocks of text content isn't really best practice, but I solved it by adding some bottom padding to the body.

Comment (nrobinson2000) [inline index.html]: I'm not as seasoned with CSS as you clearly are. :smile:

**Comment by nrobinson2000** [comment]
Awesome!

**Comment by woubuc** [comment]
I'm personally not really a fan of Java, seems a bit overkill for just some html validation, but I've merged it in for now :smiley:

**Comment by nrobinson2000** [comment]
Same. But having a nice green build passing badge is worth it.

**Comment by woubuc** [comment]
Unfortunately, we're not gonna get a green build... Apparently the Google Fonts url is not valid, according to this test.
```
file:/home/travis/build/woubuc/1pr/./index.html":9.2-9.99: error: 
Bad value "https://fonts.googleapis.com/css?family=Open+Sans:400,600|Patua+One"
for attribute "href" on element "link": Illegal character in query: "|" is not allowed.
```

**Comment by nrobinson2000** [comment]
:confused: What? That character isn't illegal.  I think its just being picky and would prefer `%7C`

**Comment by nrobinson2000** [comment]
It likes that but there's still a lonesome `li` somewhere.

```
WARNING:html5validator.validator:"file:/home/travis/build/woubuc/1pr/./index.html":30.99-30.103: error: No "li" element in scope but a "li" end tag seen.
```

**Comment by nrobinson2000** [comment]
Green!

**Comment by woubuc** [comment]
Victory! :tada:

**Comment by nrobinson2000** [comment]
Should the build passing badge be added to the footer?

**Comment by woubuc** [comment]
I think we should wait for the next PR to add the badge to the site, so @benhalpern has time to enable travis for this repo (otherwise we'll just get a 'build unknown' badge).

**Comment by DominicBisset** [comment]
Tangential question:
Is the meta for this to build every feature into one single monolithic Pull Request that can't be ignored?  I was pushing for smaller, individual-feature-focused in the comments for PR #13 but was I wrong in this?

**Comment by nrobinson2000** [comment]
Well this doesn't break @benhalpern's rules and it allows us to contribute more work.

**Comment by dakebl** [comment]
You have a good point @DominicBisset in regards to the bloated PR. 
It's something to be discussed I'm sure but I would suggest that this invokes the dev.to community ethos quite well. We all come together to shape the PR and affect it's outcome :)

**Comment by DominicBisset** [comment]
It does do both of those things.  I'm happy to be outvoted if that's the wind is blowing.  

I guess we'll learn @benhalpern thoughts on the matter tomorrow.

**Comment by dakebl** [comment]
We will need to address the rate limit of the api:

``{
  "message": "API rate limit exceeded for IP-ADDRESS. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
  "documentation_url": "https://developer.github.com/v3/#rate-limiting"
}``

**Comment by nrobinson2000** [comment]
Where are you seeing that message?

**Comment by nrobinson2000** [comment]
@woubuc I think I'm going to have to vote for https://github.com/thepracticaldev/1pr/pull/24 as the next pull request.

**Comment by srebalaji** [comment]
@woubuc Thanks for merging my PR. :)

**Comment by dakebl** [comment]
@nrobinson2000 - I got that message by opening the api link directly: https://api.github.com/repos/thepracticaldev/1pr/pulls?state=closed

**Comment by woubuc** [comment]
@dakebl regarding the rate limiting, you can create a personal access token and use that to authenticate on your end while testing. I don't know if it's secure to publish a personal token on the client side, or if there's another way to solve it, but it's definitely something that should be looked into at some point.

**Comment by dakebl** [comment]
@woubuc - Yeah, I'm not sure what can be done about it really. 

The suggestion made by GitHub is to add conditional headers to the requests that check for changes since the last request was made. If no change is found, the cached result would be passed without applying a hit to the API rate limit. Otherwise a new request is sent and the API hit logged.

**Comment by woubuc** [comment]
Yeah, I was already thinking yesterday about implementing a simple caching (e.g. with localStorage), but I didn't get around to it just yet. I probably will add that in one of these days.

**Comment by dakebl** [comment]
Hey @woubuc , in light of discussion around the tracking, would you consider merging https://github.com/thepracticaldev/1pr/pull/24 into this PR?

**Comment by woubuc** [comment]
Sure! I'm not at my pc at the moment, but if you could make a PR on my fork with your changes I'll merge them in.

**Comment by dakebl** [comment]
Done: https://github.com/woubuc/1pr/pull/1

**Comment by woubuc** [comment]
Done! #24 is merged into this PR as well.

**Comment by woubuc** [comment]
This PR is superseded by #27.

**Comment by benhalpern** [comment]
@DominicBisset your concern is accurate, but I'm not going to stand in the way of some organized progress. In the long run, it will probably be harder to pull this stuff off, but for now this will be okay and I'll try to keep on top of these things.

Feel free to open issues about these sorts of things.

**Comment by benhalpern** [comment]
Merging this request. Having trouble with gitter, but I'll update with more info. [dev.to post on the subject](https://dev.to/ben/having-trouble-integrating-gitter-into-1pr)

**Pull request merged**